test/cypress/integration/group-chat-message.spec.js

```diff
@@ -0,0 +1,210 @@
+    })
+  }
+
+  it(`user1 creats '${groupName}' group and joins "${CHATROOM_GENERAL_NAME}" channel by default`, () => {
+    cy.visit('/')
+    cy.giSignup(user1)
+    me = user1
+
+    cy.giCreateGroup(groupName, { bypassUI: true })
+    cy.giGetInvitationAnyone().then(url => {
+      invitationLinkAnyone = url
+    })
+    cy.getByDT('groupChatLink').click()
+    cy.getByDT('channelName').should('contain', CHATROOM_GENERAL_NAME)
+    cy.getByDT('channelsList').within(() => {
+      cy.get('ul').children().should('have.length', 1)
+    })
+    checkIfJoined(CHATROOM_GENERAL_NAME)
+
+    cy.giLogout()
+  })
+
+  it(`user2 joins ${groupName} group`, () => {
+    cy.giAcceptGroupInvite(invitationLinkAnyone, {
+      username: user2,
+      groupName: groupName,
+      shouldLogoutAfter: false,
+      bypassUI: true
+    })
+    me = user2
+    cy.getByDT('groupChatLink').click()
+
+    cy.getByDT('channelName').should('contain', CHATROOM_GENERAL_NAME)
+    checkIfJoined(CHATROOM_GENERAL_NAME)
+
+    cy.getByDT('channelsList').find('ul>li:first-child').within(() => {
+      cy.get('[data-test]').should('contain', CHATROOM_GENERAL_NAME)
+    })
+  })
+
+  it('user2 sends greetings and asks to have meeting this morning', () => {
+    sendMessage(`Hello ${user1}. How are you? Thanks for inviting me to this awesome group.`)
+    sendMessage('Can we have a meeting this morning?')
+  })
+
+  it('user1 sends greetings and edits', () => {
+    switchUser(user1)
+    cy.getByDT('groupChatLink').click()
+
+    sendMessage('Hi')
+
+    editMessage(6, `Hi ${user2}. I am fine thanks.`)
+  })
+
+  it('user2 edits and deletes message', () => {
+    switchUser(user2)
+    cy.getByDT('groupChatLink').click()
+
+    editMessage(5, 'Can we have a meeting this evening?')
+
+    deleteMessage(4, 4)
+  })
+
+  it('user1 sees the edited message but he is not able to see the deleted message', () => {
+    switchUser(user1)
+    cy.getByDT('groupChatLink').click()
+
+    cy.getByDT('conversationWapper').within(() => {
+      cy.get('.c-message').should('have.length', 4)
+
+      cy.get('.c-message').eq(0).find('.c-who > span:first-child').should('contain', user1)
+      cy.get('.c-message').eq(0).find('.c-text').should('contain', `Joined ${CHATROOM_GENERAL_NAME}`)
+
+      cy.get('.c-message').eq(1).find('.c-who > span:first-child').should('contain', user2)
+      cy.get('.c-message').eq(1).find('.c-text').should('contain', `Joined ${CHATROOM_GENERAL_NAME}`)
+
+      cy.get('.c-message').eq(2).find('.c-who > span:first-child').should('contain', user2)
+      cy.get('.c-message').eq(2).find('.c-text').should('contain', 'Can we have a meeting this evening?')
+
+      cy.get('.c-message').eq(3).find('.c-who > span:first-child').should('contain', user1)
+      cy.get('.c-message').eq(3).find('.c-text').should('contain', `Hi ${user2}. I am fine thanks.`)
```

Comment:
Couldn't we reduce code duplication and line lengths by defining some other helper functions, for example `checkSender()`, `checkMessage()`, or something similar?

Reply:
Correct. I will update.